python/cudf/cudf/core/reshape.py

```diff
@@ -642,67 +658,79 @@ def get_dummies(
     if drop_first:
         raise NotImplementedError("drop_first is not supported yet")
 
-    encode_fallback_dtypes = ["object", "category"]
+    if isinstance(df, cudf.DataFrame):
+        encode_fallback_dtypes = ["object", "category"]
 
-    if columns is None or len(columns) == 0:
-        columns = df.select_dtypes(include=encode_fallback_dtypes).columns
+        if columns is None or len(columns) == 0:
+            columns = df.select_dtypes(include=encode_fallback_dtypes).columns
 
-    def length_check(obj, name):
-        if cudf.utils.dtypes.is_list_like(obj):
-            if len(obj) != len(columns):
-                raise ValueError(
-                    f"Length of '{name}' ({len(obj)}) did not match the "
-                    f"length of the columns being encoded ({len(columns)})."
-                )
+        def length_check(obj, name):
+            if cudf.utils.dtypes.is_list_like(obj):
+                if len(obj) != len(columns):
+                    raise ValueError(
+                        f"Length of '{name}' ({len(obj)}) did not match the "
+                        f"length of the columns being "
+                        f"encoded ({len(columns)})."
+                    )
```

Comment:
nitpick: I don't like defining functions inline like this that really don't need to be defined inline. Feels like a waste of redefining each time `get_dummies` is called.

Reply:
Moved this to a separate internal api